src/RealMeService.php

```diff
@@ -480,7 +480,8 @@ public function enforceLogin(HTTPRequest $request, $backUrl = null)
             Member::singleton()->extend("onRealMeLoginFailure", $e);
 
             // No auth data or failed to decrypt, enforce login again
-            $this->getAuth()->login(Director::absoluteBaseURL());
+            $validBackURL = $this->validSiteURL($backUrl ?: $this->getBackURL($request));
```

Comment:
The use of  `?:` here is a bit messy, would be cleaner to change the `if ($backUrl)` block above  in this file to have an `else { $backUrl = $this->getBackURL($request) }`
Below that if/else then do the `$this->validateSiteURL()` check and fallback to `Director::absoluteBaseURL()`